virtcontainers/fc.go

```diff
@@ -138,10 +147,41 @@ func (fc *firecracker) createSandbox(ctx context.Context, id string, hypervisorC
 	//TODO: check validity of the hypervisor config provided
 	//https://github.com/kata-containers/runtime/issues/1065
 	fc.id = id
-	fc.socketPath = filepath.Join(runStoragePath, fc.id, fireSocket)
+	fc.state.set(notReady)
+
+	// unix domain socket names have a hard limit
+	// #define UNIX_PATH_MAX   108
+	// Keep it short and live within the jailer expected paths
+	// <chroot_base>/<exec_file_name>/<id>/
+	// Also jailer based on the id implicitly sets up cgroups under
+	// <cgroups_base>/<exec_file_name>/<id>/
+	hypervisorName := filepath.Base(hypervisorConfig.HypervisorPath)
```

Comment:
Please can you use [`BuildSocketPath`](https://github.com/kata-containers/runtime/blob/master/virtcontainers/utils/utils.go#L197) here.